### PR TITLE
[Home Page Picker] Adds template to the site creation analytics if a template was used

### PR DESCRIPTION
**Pull request by antonis**
Add's the template to the site creation analytics calls.

(aligns with iOS https://github.com/wordpress-mobile/WordPress-iOS/pull/15377)

### Skip on site creation
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1.  Select `Skip` on `Choose a Design`
1. Complete the flow
1. **Expect** to see the events
    - `enhanced_site_creation_site_design_preview_viewed`
    - `site_created`
    - `enhanced_site_creation_site_design_preview_loading`
    - `enhanced_site_creation_site_design_preview_loaded`

### Select site design
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1.  Select `Skip` on `Choose a Design`
1. Complete the flow
1. **Expect** to see the events
    - `enhanced_site_creation_site_design_preview_viewed` with the property `template` populated with the selected template's `slug`
    - `site_created`
    - `enhanced_site_creation_site_design_preview_loading` with the property `template` populated with the selected template's `slug`
    - `enhanced_site_creation_site_design_preview_loaded` with the property `template` populated with the selected template's `slug`

## Additional Context: 

This is a continuation of the work done in the [Home Page Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/labels/Project%3A%20Home%20Page%20Picker). 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt [hunk 1]
```diff
@@ -102,24 +91,36 @@ class SiteCreationTracker @Inject constructor(val tracker: AnalyticsTrackerWrapp
...
-        )
-    }
-
-    fun trackSiteDesignPreviewLoaded(template: String) {
-        tracker.track(
-                AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_SITE_DESIGN_PREVIEW_LOADED,
-                mapOf("template" to template)
-        )
+    fun trackSiteDesignPreviewViewed(template: String?) {
+        if (template == null || template == defaultTemplateSlug) {
```
Comment (Tug) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt:95]: does that mean we'll never have a `template: default` prop for those events? Makes sense to me as the slug for default could change later on 👍

**Review by Tug — APPROVED** [review]
LGTM 👍

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt (hunk 1 above)
Comment (antonis) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt:95]: Thank you for reviewing this @Tug 🙇 

> does that mean we'll never have a template: default prop for those events?

Exactly! I aligned with iOS on this. It seemed reasonable not to pass `default`

<image src="https://user-images.githubusercontent.com/304044/100235624-51f95900-2f35-11eb-9840-12be93d6028b.png"/>

**Comment by antonis** [comment]
Below are the analytics emitted on the site creation flow.

## Select and Preview design before choosing
```
enhanced_site_creation_accessed
enhanced_site_creation_site_design_viewed
enhanced_site_creation_site_design_preview_viewed, Properties: {"template":"alves"}
enhanced_site_creation_site_design_preview_loading, Properties: {"template":"alves"}
enhanced_site_creation_site_design_preview_loaded, Properties: {"template":"alves"}
enhanced_site_creation_site_design_selected, Properties: {"template":"alves"}
enhanced_site_creation_domains_accessed
enhanced_site_creation_domains_selected, Properties: {"chosen_domain":"antonistest53.wordpress.com","search_term":"antonistest53"}
enhanced_site_creation_site_design_preview_viewed, Properties: {"template":"alves"}
site_created
enhanced_site_creation_site_design_preview_loading, Properties: {"template":"alves"}
enhanced_site_creation_site_design_preview_loaded, Properties: {"template":"alves"}
enhanced_site_creation_preview_ok_button_tapped
```

## Select and Choose design
```
enhanced_site_creation_accessed
enhanced_site_creation_site_design_viewed
enhanced_site_creation_site_design_selected, Properties: {"template":"alves"}
enhanced_site_creation_domains_accessed
enhanced_site_creation_domains_selected, Properties: {"chosen_domain":"antonistest51.wordpress.com","search_term":"antonistest51"}
enhanced_site_creation_site_design_preview_viewed, Properties: {"template":"alves"}
site_created
enhanced_site_creation_site_design_preview_loading, Properties: {"template":"alves"}
enhanced_site_creation_site_design_preview_loaded, Properties: {"template":"alves"}
enhanced_site_creation_preview_ok_button_tapped
```

## Skip design selection
```
enhanced_site_creation_accessed
enhanced_site_creation_site_design_viewed
enhanced_site_creation_site_design_skipped
enhanced_site_creation_domains_accessed
enhanced_site_creation_domains_selected, Properties: {"chosen_domain":"antonistest54.wordpress.com","search_term":"antonistest54"}
enhanced_site_creation_site_design_preview_viewed
site_created
enhanced_site_creation_site_design_preview_loading
enhanced_site_creation_site_design_preview_loaded
enhanced_site_creation_preview_ok_button_tapped
```

**Comment by antonis** [comment]
Closing this in favor of https://github.com/wordpress-mobile/WordPress-Android/pull/13489

**Pull request closed without merging**